pilot/pkg/networking/plugin/plugin.go

```diff
@@ -47,8 +47,12 @@ const (
 )
 
 // ModelProtocolToListenerProtocol converts from a config.Instance to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(p protocol.Instance) ListenerProtocol {
-	switch p {
+func ModelProtocolToListenerProtocol(p *protocol.Instance) ListenerProtocol {
+	if *p == protocol.Unsupported {
```

Comment:
This seems hacky.  I don't think this should be done here. If the caller wants to do this, it should be done elsewhere. I'm not sure that this should even be included in this PR.

Reply:
Move this code to the caller side.